todoapp-main/src/main/java/ru/todoapp/service/TaskService.java

```diff
@@ -0,0 +1,175 @@
+        } else if (Instant.now().isAfter(Instant.parse(addTaskRequestDTO.getDatetime()))) {
+            sendRequestResultDTO(addTaskRequestDTO.getRequestUUID(), RequestStatus.FAIL,
+                    "Can't create task in past!");
+        } else {
+            taskRepository.saveNewTask(SaveTaskEntity.of(addTaskRequestDTO));
+            sendRequestResultDTO(addTaskRequestDTO.getRequestUUID(), RequestStatus.SUCCESS, null);
+        }
+
+        requestRepository.save(RequestEntity.of(addTaskRequestDTO));
+    }
+
+    /**
+     * Task fetching handler
+     */
+    public void handleFetching(FetchTasksRequestDTO fetchTasksRequestDTO) {
+        List<String> unfilled = allUnfilledFields(fetchTasksRequestDTO);
+        if (!unfilled.isEmpty()) {
+            sendRequestResultDTO(fetchTasksRequestDTO.getRequestUUID(),
+                    RequestStatus.FAIL,
+                    "Fields: " + String.join(", ", unfilled) + " are empty!");
+        } else if (!userRepository.exists(fetchTasksRequestDTO.getUserUUID())) {
+            sendRequestResultDTO(fetchTasksRequestDTO.getRequestUUID(), RequestStatus.FAIL,
+                    "Can't find user!");
+        } else if (!DateTimeUtils.isCorrectDateFormat(fetchTasksRequestDTO.getBeginDate())
+                || !DateTimeUtils.isCorrectDateFormat(fetchTasksRequestDTO.getEndDate())) {
+            sendRequestResultDTO(fetchTasksRequestDTO.getRequestUUID(), RequestStatus.FAIL,
+                    "Incorrect search filter: " + incorrectDateOrDates(fetchTasksRequestDTO)
+                            + " have incorrect format!");
+        } else if (DateTimeUtils.setAtTheStartOfTheDay(fetchTasksRequestDTO.getBeginDate())
+                .isAfter(DateTimeUtils.setAtTheEndOfTheDay(fetchTasksRequestDTO.getEndDate()))) {
+            sendRequestResultDTO(fetchTasksRequestDTO.getRequestUUID(), RequestStatus.FAIL,
+                    "Incorrect search filter: begin date is after end date!");
+        } else {
+            List<TaskEntity> tasksEntities = taskRepository.getTasksList(GetTaskRequestEntity.of(fetchTasksRequestDTO));
+            List<TaskDTO> taskDTOS = convertTaskEntityListTypeToDTO(tasksEntities);
+            var result = new FetchTasksResponseDTO(
+                    fetchTasksRequestDTO.getRequestUUID(),
+                    fetchTasksRequestDTO.getUserUUID(),
+                    taskDTOS);
+            sendRequestResultDTO(fetchTasksRequestDTO.getRequestUUID(), RequestStatus.SUCCESS,
+                    null);
+            fetchTasksResponseDTOKafkaTemplate.send(KafkaTopics.FETCH_RESPONSE_TASKS_TOPIC, result);
+        }
+
+        requestRepository.save(RequestEntity.of(fetchTasksRequestDTO));
+    }
+
+    /**
+     * creates a list of all unfilled fields for adding tasks
+     *
+     * @param addTaskRequestDTO - the object checked for unfilled fields
+     */
+    private List<String> allUnfilledFields(AddTaskRequestDTO addTaskRequestDTO) {
```

Comment:
1. better name it like `getAllUnfilledFields`, it's a method, not a field
2. as i can see you use those strings for messaging, but it's not necessarily. You can override `toString()` method in our class and bypass object to the message. Overriden method does all job for you.
3. To check whatever it's all field are fulfilled i suggest to create some kind of DTO validator, or better yet create abstract boolean method in `RequestDTO`, implemented in each DTO that returns all fields are valid or not. That way, we don't new boilerplate overriden methods in the future.

Reply:
I think we shouldn't spend time on DTO validators right now, 'cause we can use hibernate validators instead. This validators (without hibernate) are only for testing purposes.

Reply:
Cute enthusiasm, but ok. That said, only point 1 remains from my comment.

Reply:
I did not quite get what do you mean by overriding `toString()` to return an empty field names. If I understood you correctly I will create a new method in the classes that will be `getAllUnfilledFields()` that will return a list of Strings containing names of the fields, I dont think I need to override the `toString()` for that... If that is not what you meant I quite dont understand how `toString()`, that will return `String` containing lets say every field value of an object, will help me in determining the missing ones, I believe it will be quite a pain in the ass to work with that String value

Reply:
`toString()` has nothing to do with determining empty fields. This thing helps you to avoid verbose passing name fields, thats it. Ovveride object `toString()`, then call it in `sendRequestResultDTO` for example, in message argument. Profit.

Reply:
I think I get it, but on second reread of your first comment I think we are talking about different things

You suggest `toString()` method to avoid:
`var result = SomethingDTO.builder()...
SomethingDTOKafkaTemplate.send(..., result);`
Correct?

I was thinking of that suggestion on the `allUnfilledFields()` method, that method only for creation of failed request in case fail accured then one or more fields are empty